### PR TITLE
Slide animations always use CSS3 transitions instead of .animate()

### LINKED ISSUE #46563: [Enhancement]: FlexSlider can always use CSS3 transitions for slide animations
### Describe the solution you'd like

The slide FlexSlider animations use either

1. [Element.animate()](https://developer.mozilla.org/en-US/docs/Web/API/Element/animate) when `video` is set to `true` or `useCSS` is set to `false`
2. CSS3 transitions with default options (which are `video` set to `false` and `useCSS` set to `true`

FlexSlider can easily use CSS3 transitions in both cases with no apparent issues, and that would simplify the complexity of the animation engine.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by apenchev**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #46563.

### How to test the changes in this Pull Request:

1. Add this custom code to your theme's `functions.php` in order to enable fade animations:
```
function carousel_options( $options ) {
	return array_merge(
		$options,
		array(
			'video' => true,
			'animationSpeed' => 1600,
			'easing' => 'easeInOutBack',
		)
	);
}
add_filter( 'woocommerce_single_product_carousel_options', 'carousel_options' );
```
2. Visit a Single Product Page.
3. Confirm that switching between gallery images happens with a slide transition that lasts 1.6 seconds.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message

Reduced the number of FlexSlider animation engines from 2 to 1, now always using CSS3 transitions.

#### Comment 

</details>

### CODE REVIEW

**Review by nefeline — APPROVED** [review]
Thank you so much for your contribution @apenchev !

I've tested and confirmed that it is possible to control the animationSpeed via the filter as expected.

I'm pre-approving, but also tagging @roykho and @thealexandrelara for additional review since they have recently worked with the gallery block and might have additional input to offer.

File: plugins/woocommerce/client/legacy/js/flexslider/jquery.flexslider.js [hunk 1]
```diff
@@ -824,16 +810,17 @@
...
 
+      dur = (dur !== undefined) ? (dur/1000) + "s" : "0s";
+      slider.container.css("transition-duration", dur);
+
       if (slider.transitions) {
         target = (vertical) ? "translate3d(0," + target + ",0)" : "translate3d(" + (parseInt(target)+'px') + ",0,0)";
-        dur = (dur !== undefined) ? (dur/1000) + "s" : "0s";
-         slider.container.css("transition-duration", dur);
+      } else {
+        slider.container.css("transition-timing-funciton", easing);
```
Comment (thealexandrelara) [inline plugins/woocommerce/client/legacy/js/flexslider/jquery.flexslider.js]: I believe there is a small typo in here it should be `transition-timing-function`, right?

```suggestion
        slider.container.css("transition-timing-function", easing);
```

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
Hey @apenchev,
Thanks a lot for working on this! Your contribution is very much appreciated.

I noticed there's a small typo in the `transition-timing-function` CSS property. This causes the `easing` function not to be applied when `slider.transitions` is set to false. Once you fix this, I can take another look and approve the PR.

## With the typo | video: `true` | easing: `easeInOutBack` |

https://github.com/woocommerce/woocommerce/assets/20469356/d31d324b-534c-4847-8477-e20e0fb7e729

## Without the typo | video: `true` | easing: `easeInOutBack` |

https://github.com/woocommerce/woocommerce/assets/20469356/d270d186-0fb1-4f36-830b-fda8de85ca84

---

> I'm pre-approving, but also tagging @roykho and @thealexandrelara for additional review since they have recently worked with the gallery block and might have additional input to offer.

Thanks for the ping, @nefeline! I haven't directly worked on the jQuery slider myself, but it's good to know how things work behind the scenes 😄.

**Comment by nefeline** [comment]
@thealexandrelara

> I noticed there's a small typo in the transition-timing-function CSS property.

Nice catch! 🙌

File: plugins/woocommerce/client/legacy/js/flexslider/jquery.flexslider.js (hunk 1 above)
Comment (apenchev) [reply plugins/woocommerce/client/legacy/js/flexslider/jquery.flexslider.js:819]: Fix typo.

**Comment by apenchev** [comment]
> Hey @apenchev, Thanks a lot for working on this! Your contribution is very much appreciated.
> 
> I noticed there's a small typo in the `transition-timing-function` CSS property. This causes the `easing` function not to be applied when `slider.transitions` is set to false. Once you fix this, I can take another look and approve the PR.
> 
> ## With the typo | video: `true` | easing: `easeInOutBack` |
>  CleanShot.2024-05-02.at.12.24.43.mp4 
> ## Without the typo | video: `true` | easing: `easeInOutBack` |
>  CleanShot.2024-05-02.at.12.25.47.mp4 
> > I'm pre-approving, but also tagging @roykho and @thealexandrelara for additional review since they have recently worked with the gallery block and might have additional input to offer.
> 
> Thanks for the ping, @nefeline! I haven't directly worked on the jQuery slider myself, but it's good to know how things work behind the scenes 😄.

Excellent catch, thanks for checking! Commited the typo fix.

**Review by thealexandrelara — APPROVED** [review]
Awesome! Thank you for fixing that! I'm approving the PR ✅

**Comment by thealexandrelara** [comment]
There's one thing I'd like you to do, though. Could you please use `pnpm --filter=./plugins/woocommerce changelog add` to add a changelog entry? This way, the required checks will pass and we can merge the PR.

**Comment by apenchev** [comment]
@thealexandrelara Added a changelog.

**Pull request merged**